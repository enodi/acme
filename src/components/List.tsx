import React from "react";

import ListItem from "./ListItem";

const List: React.FC<{
  userData: any;
  toggleConversationType: any;
}> = ({ userData, toggleConversationType }) => (
  <div>
    {userData.length > 0 &&
      userData.map((item: any) => (
        <ListItem
          key={`list-item-${item.id}`}
          item={item}
          toggleConversationType={toggleConversationType}
        />
      ))}
  </div>
);

export default List;
