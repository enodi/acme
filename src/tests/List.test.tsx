import React from "react";
import { shallow } from "enzyme";

import List from "../components/List";
import { selectedItem } from "./fixtures";

let toggleConversationType, handleSelected, wrapper: any;
beforeEach(() => {
  toggleConversationType = jest.fn();
  handleSelected = jest.fn();
  wrapper = shallow(
    <List
      toggleConversationType={toggleConversationType}
      userData={[selectedItem]}
      handleSelected={handleSelected}
    />
  );
});

test("should render List component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
