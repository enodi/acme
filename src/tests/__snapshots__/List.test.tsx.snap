// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render List component correctly 1`] = `
<div>
  <ListItem
    handleSelected={[MockFunction]}
    item={
      Object {
        "email": "krotte0@webmd.com",
        "firstName": "Kirby",
        "id": 1,
        "isFavorite": true,
        "isSelected": true,
        "lastName": "Rotte",
        "messages": Array [
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        ],
      }
    }
    key="list-item-1"
    toggleConversationType={[MockFunction]}
  />
</div>
`;
